analytics/service.go

```diff
@@ -0,0 +1,142 @@
+package analytics
+
+import (
+	"fmt"
+	"os"
+	"sync"
+	"time"
+
+	r "github.com/rudderlabs/analytics-go/v4"
+)
+
+var (
+	Instance *Service
+	once     sync.Once
+)
+
+func Init(config Config) {
+	once.Do(func() {
+		Instance = newService(config)
+	})
+}
+
+func GetService() *Service {
+	return Instance
+}
+
+type Service struct {
+	rudderClient r.Client
+}
+
+type Config struct {
+	WriteKey  string
+	DataPlane string
+	Verbose   bool
+}
+
+func newService(config Config) *Service {
+	if config.WriteKey == "" || config.DataPlane == "" {
+		fmt.Printf("WARN: Rudderstack is not configured, data will not be sent to Rudderstack")
+	}
+
+	c, err := r.NewWithConfig(config.WriteKey,
+		r.Config{
+			DataPlaneUrl: config.DataPlane,
+			Interval:     1 * time.Second,
+			BatchSize:    100,
+			Verbose:      config.Verbose,
+			DisableGzip:  false,
+		})
+
+	if err != nil {
+		fmt.Printf("FATAL: Failed to create rudderstack client: %v", err)
+		panic(err)
```

Comment:
The warning in connection with the panic below does not make sense. We should have on or the other. Considering this is fringe functionality and if we break it I would prefer not to have the analytics data as compared to not having the system running.

So please remove the panic, and then in each of the "*Started", "*Ended" functions check for `s.rudderClient == nil` and debug log the info.